include/fixed_containers/reflection.hpp

```diff
@@ -42,56 +44,63 @@ class DepthTracker
     }
 };
 
-template <typename FieldTypeNameStringType = fixed_string_detail::FixedString<128>,
-          typename FieldNameStringType = fixed_string_detail::FixedString<64>>
-class FieldInfo
+class FieldEntry
 {
 private:
-    FieldTypeNameStringType field_type_name_{};
-    FieldNameStringType field_name_{};
+    std::string_view field_type_name_;
+    std::string_view field_name_;
+    int depth_;
 
 public:
-    constexpr FieldInfo() noexcept
-      : field_type_name_{}
-      , field_name_{}
-    {
-    }
-
-    constexpr FieldInfo(const std::string_view& field_type_name,
-                        const std::string_view& field_name) noexcept
+    constexpr FieldEntry(const std::string_view& field_type_name,
+                         const std::string_view& field_name,
+                         const int depth) noexcept
       : field_type_name_{field_type_name}
       , field_name_{field_name}
+      , depth_{depth}
     {
     }
 
 public:
     [[nodiscard]] constexpr std::string_view field_type_name() const { return field_type_name_; }
     [[nodiscard]] constexpr std::string_view field_name() const { return field_name_; }
+    [[nodiscard]] constexpr int depth() const { return depth_; }
 };
 
-template <typename T>
-constexpr std::size_t field_count_of(const T& instance)
+template <typename T, typename Func>
```

Comment:
nit: we can use the `std::invocable` concept to make the functional argument a little more typesafe and get better diagnostics.

```cpp
template <typename T, std::invocable<FieldEntry> Func>
constexpr void for_each_field_entry(const T& instance, Func func)
{
    ...
}
```

Reply:
Done in: https://github.com/teslamotors/fixed-containers/pull/56